experiment/secret-sync-controller/main_test.go

```diff
@@ -1,295 +1,435 @@
 package main
 
 import (
+	"bytes"
+	"context"
 	"fmt"
-	"reflect"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/test-infra/experiment/secret-sync-controller/client"
 	"testing"
 )
 
-type clientMock struct { // mock client
-	K8sSecret           map[string]map[string]map[string][]byte
-	SecretManagerSecret map[string]map[string][]byte
-}
+/*
+#testing-setup
 
-func (cl clientMock) GetKubernetesSecret(spec KubernetesSpec) (*SecretData, error) {
-	_, ok := cl.K8sSecret[spec.Namespace][spec.Secret]
-	if !ok {
-		return nil, fmt.Errorf("Secret not found: K8s:/namespaces/%s/secrets/%s", spec.Namespace, spec.Secret)
-	}
-	val, ok := cl.K8sSecret[spec.Namespace][spec.Secret][spec.Key]
-	if !ok {
-		return new(SecretData), nil
-	}
-	return &SecretData{val}, nil
-}
-func (cl clientMock) GetSecretManagerSecret(spec SecretManagerSpec) (*SecretData, error) {
-	val, ok := cl.SecretManagerSecret[spec.Project][spec.Secret]
-	if !ok {
-		return nil, fmt.Errorf("Secret not found: SecretManager:/projects/%s/secrets/%s", spec.Project, spec.Secret)
-	}
-	return &SecretData{val}, nil
-}
+secretManager:
+  project:
+  - name: k8s-jkns-gke-soak
+    secret:
+    - name: gsm-password
+      data: gsm-password-v1
+    - name: gsm-token
+      data: gsm-token-v1
+kubernetes:
+  namespace:
+  - name: ns-a
+    secret:
+    - name: secret-a
+      key:
+      - name: key-a
+        data: old-token
+  - name: ns-b
+    secret:
+    - name: secret-a
+  - name: ns-c
```

Comment:
Note you could do something a bit more concise if you assume duplicate names aren't allowed to exist
```yaml
secretManager:
  k8s-jkns-gke-soak:
    gsm-password: gsm-password-v1
    gsm-token: gsm-token-v1
kubernetes:
  ns-a:
    secret-a:
      key-a: old-token
  ns-b:
    secret-b: {}
```

Reply:
In this concise form, I think it's necessary to use `json.Unmarshal` instead.
`yaml.Unmarshal` always yields type of `map[interface{}]interface{}`, which I assume is a red flag.

With `json.Unmarshal`, we can just use `map[string]interface{}`.